### PR TITLE
fix: reduce incorrectly for group-by with offset(#30828)

### PR DESCRIPTION
**Pull request by MrPresent-Han**
related: #30828

### CODE REVIEW

**Review by MrPresent-Han — COMMENTED** [review]
review

File: internal/proxy/search_reduce_util.go [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+	"github.com/milvus-io/milvus/pkg/log"
+	"github.com/milvus-io/milvus/pkg/util/merr"
+	"github.com/milvus-io/milvus/pkg/util/metric"
+	"github.com/milvus-io/milvus/pkg/util/paramtable"
+	"github.com/milvus-io/milvus/pkg/util/timerecord"
+	"github.com/milvus-io/milvus/pkg/util/typeutil"
+)
+
+type reduceSearchResultInfo struct {
+	ctx                 context.Context
```
Comment (congqixia) [inline internal/proxy/search_reduce_util.go:22]: NIT: do not store ctx in struct, it is not recommended by Golang convention

Comment (MrPresent-Han) [reply internal/proxy/search_reduce_util.go]: changed

**Review by congqixia — APPROVED** [review]
/lgtm

**Comment by congqixia** [comment]
overall logic lgtm, but the got confused by introducing `NewReduceSearchResultInfo` struct, it just stores the info and not method are related to it

**Pull request merged**